model/AdyenNotification.php

```diff
@@ -213,4 +213,17 @@ public function isDuplicate($notification)
 
         return $this->dbInstance->getValue($sql);
     }
+
+    /**
+     * @param $merchantReference
+     * @return mixed
+     */
+    public function getProcessedNotificationsByMerchantReference($merchantReference)
+    {
+        $sql = 'SELECT * FROM ' . _DB_PREFIX_ . self::$tableName
+            . ' WHERE `merchant_reference` = "' . pSQL($merchantReference) . '"'
+            . ' AND `done` = 1';
+
+        return $this->dbInstance->executeS($sql);
+    }
```

Comment:
We can make it a static method easier to call without the need to declare the model.

Reply:
It is easier to mock the function if it is not static and when we created the notification model we decided to not create static functions even though almost all the functions in that class could be static.

Reply:
Static methods are harder to test in PHP. Also, PrestaShop offers a Dependency Injection suite so it's not necessary to declare the model manually.

Reply:
Ok, clear.